### DISCUSSION TITLE
Zatca Signed properties

### DISCUSSION THREAD

**Discussion by EhabHamed2000** [mrsool/zatca#31] [Q&A]
I had read your conversation in [Zatca Developer Community](https://zatca1.discourse.group/t/what-do-signed-properties-look-like-when-hashing/717) but I didn't know how do you manage to solve the problem. Can you please tell me?

**Comment by obahareth** [comment]
Hey @EhabHamed2000, if you use this SDK it already takes care of this part for you.

There are 2 things you must make sure of:
1. The whole signed properties block must be indented EXACTLY like ZATCA's examples when you sign it
2. It must also be indented the same way in the Base64-encoded XML you upload to ZATCA.

In our SDK we made sure that we normalize to the same spaces ZATCA expects whenever we hash, but on ZATCA's serverside they don't this, they will simply hash it with whatever whitespace you provided.

So if you hash it correctly but send them the signed properties with an extra space somewhere, it will fail because when they hash it on their end the extra space will result in a different hash.


tl;dr version:
Ensure you are matching ZATCA's spaces and newlines without any change. Spaces must be identical to ZATCA.
